test/runtests.jl

```diff
@@ -87,6 +87,18 @@ using Base: specializations
     end
     PrecompileTools.verbose[] = oldval
 
+    using NotToplevel
+    # Check that calls inside @setup_workload are not precompiled
+    m = NotToplevel.NotPrecompiled.themethod
+    have_char = have_float = false
+    for mi in specializations(m)
+        mi === nothing && continue
+        have_char |= mi.specTypes.parameters[2] === Char
+        have_float |= mi.specTypes.parameters[2] === Float64
+    end
+    @test !have_float       # not wrapped inside `@compile_workload`
+    @test_broken have_char  # is wrapped
```

Comment:
@mortenpi @benlorenz note that the pattern used in your reports blocks effective precompilation. If you haven't followed, here's the issue: if you compile external (outside of the package, e.g., Base or whatever) code before you reach `@compile_workload`, it will not be cached. The problem is that your construct forces compilation of the anonymous function before `@compile_workload` turns on. Thus, any code not "owned" by your package that can't be inferrably traced back to a caller in your package will be dropped from the precompile cache.

As a guide to understanding the test, know that `Base.inferencebarrier` is used in `NotPrecompiled` to force runtime dispatch.

Reply:
Also CC @jpthiele